mteb/load_results/benchmark_results.py

```diff
@@ -0,0 +1,339 @@
+import json
+from collections import defaultdict
+from pathlib import Path
+from typing import Any, Callable, Iterable, Literal
+
+import numpy as np
+import pandas as pd
+from pydantic import BaseModel, ConfigDict
+
+from mteb.abstasks.AbsTask import AbsTask, ScoresDict
+from mteb.abstasks.TaskMetadata import (ISO_LANGUAGE_SCRIPT, TASK_CATEGORY,
+                                        TASK_DOMAIN, TASK_TYPE)
+from mteb.languages import ISO_LANGUAGE
+from mteb.load_results.task_results import TaskResult
+from mteb.models.overview import get_model_metas
+from mteb.overview import get_tasks
+
+Split = str
+Score = Any
+
+
+def restrict_task_results(res: TaskResult, task: AbsTask) -> TaskResult:
```

Comment:
Is this a copy of existing utility? Did copy it or was it moved?

edit seems like you refactored it

Reply:
```suggestion
def _restrict_task_results(task_result: TaskResult, task: AbsTask) -> TaskResult:
```